### PR TITLE
[JN-962] i18n for static dashboard alerts

### PR DESCRIPTION
**Pull request by MatthewBemis**
#### DESCRIPTION (include screenshots, and mobile screenshots for participant UX)

This adds all of the static dashboard messages to languageTexts (i.e. Form X completed, Welcome to the study, Cookies banner, etc). It also updates the i18n function to support substitutions.

Screenshot 2024-04-30 at 1 02 01 PM

#### TO TEST:  *(simple manual steps for confirming core behavior -- used for pre-release checks)*

* Restart Participant Api App and Admin Api App (this is what will repopulate the new language texts)
* Rebuild ui-core
* Go to demo http://sandbox.demo.localhost:3001/
* Log in and complete a survey in Spanish
* Confirm that you see "\<formName\> completado" as the message when you submit the form

### CODE REVIEW

File: ui-admin/src/navbar/StudySidebar.tsx [hunk 1]
```diff
@@ -78,10 +78,6 @@ export const StudySidebar = ({ study, portalList, portalShortcode }:
           <NavLink to={studyEnvNotificationsPath(portalShortcode, study.shortcode, 'sandbox')}
             className={sidebarNavLinkClasses} style={navStyleFunc}>Emails &amp; Notifications</NavLink>
         </li>
-        <li>
-          <NavLink to={studyEnvAlertsPath(portalShortcode, study.shortcode, 'sandbox')}
-            className={sidebarNavLinkClasses} style={navStyleFunc}>Participant Dashboard</NavLink>
-        </li>
```
Comment (MatthewBemis) [inline ui-admin/src/navbar/StudySidebar.tsx:84]: We're deprecating custom dashboard messages for the moment. All of the underlying code is still there and it's still accessible at https://localhost:3000/demo/studies/heartdemo/env/sandbox/alerts

File: ui-core/src/participant/I18nProvider.tsx [hunk 2]
```diff
@@ -63,8 +68,16 @@ export function I18nProvider({ defaultLanguage, portalShortcode, children }: {
     })
   }
 
-  const i18n = (key: string, defaultValue?: string) => {
-    return languageTexts[key] || defaultValue || `{${key}}`
+  const i18n = (key: string, options?: I18nOptions) => {
+    let text = languageTexts[key] || options?.defaultValue || `{${key}}`
+    if (options && options.substitutions) {
+      Object.keys(options.substitutions).forEach(substitutionKey => {
+        if (options.substitutions) {
```
Comment (devonbush) [inline ui-core/src/participant/I18nProvider.tsx]: that's annoying that typescript requires the extra check.  consider replacing with a `for` loop
```
for (const substitutionKey in Object.keys(options.substitutions)) {
        text = substitute(text, substitutionKey, options.substitutions)
      }
```
and then 
```
  const substitute = (text: string, substitutionKey: string, substitutions: Record<string, string>): string => {
    return text.replace(`{${substitutionKey}}`, substitutions[substitutionKey])
  }
```

File: populate/src/main/resources/seed/i18n/dev/languageTexts.json [hunk 3]
```diff
@@ -67,6 +67,14 @@
   {"keyName": "suggestBetterAddressHeader", "text": "DEV_Is this the correct address?", "language": "dev"},
   {"keyName": "suggestBetterAddressBody", "text": "DEV_We were able to verify a similar address. Please select which you would like to use.", "language": "dev"},
   {"keyName": "newAddress", "text": "DEV_New Address", "language": "dev"},
-  {"keyName": "oldAddress", "text": "DEV_Old Address", "language": "dev"}
-
+  {"keyName": "oldAddress", "text": "DEV_Old Address", "language": "dev"},
+  {"keyName": "hubUpdateFormSubmittedTitle", "text": "DEV_{formName} completed", "language": "dev"},
+  {"keyName": "hubUpdateNoActivitiesTitle", "text": "DEV_All activities complete", "language": "dev"},
+  {"keyName": "hubUpdateNoActivitiesDetail", "text": "DEV_You have completed all activities for this study. We will notify you as soon as new activities are available. Thank you for your participation!", "language": "dev"},
+  {"keyName": "hubUpdateWelcomeToStudyTitle", "text": "DEV_Welcome to {studyName}", "language": "dev"},
```
Comment (devonbush) [inline populate/src/main/resources/seed/i18n/dev/languageTexts.json]: missing $ in substitution

File: populate/src/main/resources/seed/i18n/en/languageTexts.json [hunk 4]
```diff
@@ -67,5 +67,14 @@
   {"keyName": "suggestBetterAddressHeader", "text": "Is this the correct address?", "language": "en"},
   {"keyName": "suggestBetterAddressBody", "text": "We were able to verify a similar address. Please select which you would like to use.", "language": "en"},
   {"keyName": "newAddress", "text": "New Address", "language": "en"},
-  {"keyName": "oldAddress", "text": "Old Address", "language": "en"}
+  {"keyName": "oldAddress", "text": "Old Address", "language": "en"},
+  {"keyName": "hubUpdateFormSubmittedTitle", "text": "{formName} completed", "language": "en"},
```
Comment (devonbush) [inline populate/src/main/resources/seed/i18n/en/languageTexts.json]: missing $ in substitution

File: populate/src/main/resources/seed/i18n/es/languageTexts.json [hunk 5]
```diff
@@ -68,6 +67,14 @@
   {"keyName": "suggestBetterAddressHeader", "text": "¿Es esta la dirección correcta?", "language": "es"},
   {"keyName": "suggestBetterAddressBody", "text": "Pudimos verificar una dirección similar. Por favor, seleccione cuál le gustaría usar.", "language": "es"},
   {"keyName": "newAddress", "text": "Nueva Dirección", "language": "es"},
-  {"keyName": "oldAddress", "text": "Dirección Anterior", "language": "es"}
-
+  {"keyName": "oldAddress", "text": "Dirección Anterior", "language": "es"},
+  {"keyName": "hubUpdateFormSubmittedTitle", "text": "{formName} completado", "language": "es"},
```
Comment (devonbush) [inline populate/src/main/resources/seed/i18n/es/languageTexts.json]: missing $ in substitution

File: populate/src/main/resources/seed/i18n/es/languageTexts.json [hunk 6]
```diff
@@ -68,6 +67,14 @@
...
+  {"keyName": "hubUpdateNoActivitiesTitle", "text": "Todas las actividades completadas.", "language": "es"},
+  {"keyName": "hubUpdateNoActivitiesDetail", "text": "Ha completado todas las actividades de este estudio. Te avisaremos tan pronto como haya nuevas actividades disponibles. ¡Gracias por su participación!", "language": "es"},
+  {"keyName": "hubUpdateWelcomeToStudyTitle", "text": "Bienvenido a {studyName}", "language": "es"},
```
Comment (devonbush) [inline populate/src/main/resources/seed/i18n/es/languageTexts.json]: missing $ in substitution

File: ui-core/src/participant/I18nProvider.tsx [hunk 7]
```diff
@@ -63,8 +68,16 @@ export function I18nProvider({ defaultLanguage, portalShortcode, children }: {
     })
   }
 
-  const i18n = (key: string, defaultValue?: string) => {
-    return languageTexts[key] || defaultValue || `{${key}}`
+  const i18n = (key: string, options?: I18nOptions) => {
```
Comment (devonbush) [inline ui-core/src/participant/I18nProvider.tsx:75]: good call making this an options object

**Review by devonbush — APPROVED** [review]
looks good, some typos and a cleanup suggestion, nothing blocking

**Review by connorlbark — APPROVED** [review]

**Pull request merged**